src/pages/iv-opening-cost/Forms/OpeningCostForm.js

```diff
@@ -0,0 +1,152 @@
+import { Grid } from '@mui/material'
+import { useContext, useEffect } from 'react'
+import * as yup from 'yup'
+import FormShell from 'src/components/Shared/FormShell'
+import toast from 'react-hot-toast'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { useInvalidate } from 'src/hooks/resource'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { useForm } from 'src/hooks/form'
+import { ControlContext } from 'src/providers/ControlContext'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { InventoryRepository } from 'src/repositories/InventoryRepository'
+import { ResourceLookup } from 'src/components/Shared/ResourceLookup'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import CustomNumberField from 'src/components/Inputs/CustomNumberField'
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+
+export default function OpeningCostForm({ labels, maxAccess, recordId, record }) {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+
+  const invalidate = useInvalidate({
+    endpointId: InventoryRepository.OpeningCost.qry
+  })
+
+  const { formik } = useForm({
+    initialValues: {
+      recordId: null,
+      itemId: null,
+      year: '',
+      avgCost: 0
+    },
+    maxAccess,
+    enableReinitialize: true,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      year: yup.string().required(),
+      itemId: yup.string().required(),
+      avgCost: yup.number().min(0).max(999999999)
+    }),
+    onSubmit: async obj => {
+      await postRequest({
+        extension: InventoryRepository.OpeningCost.set,
+        record: JSON.stringify(obj)
+      })
+
+      if (!obj.recordId) {
+        toast.success(platformLabels.Added)
+
+        formik.setValues({
+          ...obj,
+          recordId: String(obj.year) + String(obj.itemId)
+        })
+      } else toast.success(platformLabels.Edited)
+
+      invalidate()
+    }
+  })
+
+  const editMode = !!formik.values.recordId
+
+  useEffect(() => {
+    ;(async function () {
+      if (record && record?.itemId && record?.year && recordId) {
```

Comment:
no need for `record && ` you are already checking it by adding "?" in itemId and year